src/plugins/vis_type_timeseries/public/application/components/lib/convert_series_to_datatable.ts

```diff
@@ -6,51 +6,73 @@
  * Side Public License, v 1.
  */
 import { IndexPattern } from 'src/plugins/data/public';
-import {
-  Datatable,
-  DatatableRow,
-  DatatableColumn,
-  DatatableColumnType,
-} from 'src/plugins/expressions/public';
+import { DatatableRow, DatatableColumn, DatatableColumnType } from 'src/plugins/expressions/public';
+import { Query } from 'src/plugins/data/common';
 import { TimeseriesVisParams } from '../../../types';
 import type { PanelData } from '../../../../common/types';
+import { BUCKET_TYPES } from '../../../../common/enums';
 import { fetchIndexPattern } from '../../../../common/index_patterns_utils';
 import { getDataStart } from '../../../services';
 import { X_ACCESSOR_INDEX } from '../../visualizations/constants';
+import type { TSVBTables } from './types';
 
-interface TSVBTables {
-  [key: string]: Datatable;
+interface FilterParams {
+  filter?: Query;
+  label?: string;
+  field?: string;
 }
 
 interface TSVBColumns {
   id: number;
   name: string;
-  isSplit: boolean;
+  isMetric: boolean;
+  type: string;
+  params?: FilterParams[];
 }
 
 export const addMetaToColumns = (
   columns: TSVBColumns[],
-  indexPattern: IndexPattern,
-  metricsType: string
+  indexPattern: IndexPattern
 ): DatatableColumn[] => {
   return columns.map((column) => {
     const field = indexPattern.getFieldByName(column.name);
     const type = (field?.spec.type as DatatableColumnType) || 'number';
+
+    let params: unknown = {
+      field: field?.spec.name,
+    };
+    if (column.type === BUCKET_TYPES.FILTERS && column.params) {
+      const filters = column.params.map((col) => ({
+        input: col.filter,
```

Comment:
nit: I think if you add `label: col.label` in here, it will pick it up as custom label for the filter pill (this is what happens for Lens/Visualize as well)

Reply:
You are absolutely right
![image](https://user-images.githubusercontent.com/17003240/117964735-a39e7000-b32a-11eb-8a04-aac5229c62b0.png)

Fixed https://github.com/elastic/kibana/pull/97426/commits/eb75702f27cd7652f6795cfadc5de553bcb41499